<a href="https://colab.research.google.com/github/gatimo256/project2/blob/main/Project_2_Part_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
#import libraries
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np
import seaborn as sns

In [9]:
#load dataset 1 Bank Marketing
path = 'https://drive.google.com/uc?export=download&id=1aMxQETdw7BnfMFLRAz4cmkTShyzzYy4m'
bank_df = pd.read_csv(path, delimiter =';')
bank_df.head()


age          job  marital  education default  balance housing loan  \
0   30   unemployed  married    primary      no     1787      no   no   
1   33     services  married  secondary      no     4789     yes  yes   
2   35   management   single   tertiary      no     1350     yes   no   
3   30   management  married   tertiary      no     1476     yes  yes   
4   59  blue-collar  married  secondary      no        0     yes   no   

    contact  day month  duration  campaign  pdays  previous poutcome   y  
0  cellular   19   oct        79         1     -1         0  unknown  no  
1  cellular   11   may       220         1    339         4  failure  no  
2  cellular   16   apr       185         1    330         1  failure  no  
3   unknown    3   jun       199         4     -1         0  unknown  no  
4   unknown    5   may       226         1     -1         0  unknown  no

In [8]:
bank_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4521 entries, 0 to 4520
Data columns (total 17 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   age        4521 non-null   int64 
 1   job        4521 non-null   object
 2   marital    4521 non-null   object
 3   education  4521 non-null   object
 4   default    4521 non-null   object
 5   balance    4521 non-null   int64 
 6   housing    4521 non-null   object
 7   loan       4521 non-null   object
 8   contact    4521 non-null   object
 9   day        4521 non-null   int64 
 10  month      4521 non-null   object
 11  duration   4521 non-null   int64 
 12  campaign   4521 non-null   int64 
 13  pdays      4521 non-null   int64 
 14  previous   4521 non-null   int64 
 15  poutcome   4521 non-null   object
 16  y          4521 non-null   object
dtypes: int64(7), object(10)
memory usage: 600.6+ KB


# Bank Marketing
1) Source of data

- Source:

[Moro et al., 2014] S. Moro, P. Cortez and P. Rita. A Data-Driven Approach to Predict the Success of Bank Telemarketing. Decision Support Systems, Elsevier, 62:22-31, June 2014

2) Brief description of data
- The data is related with direct marketing campaigns of a Portuguese banking institution. The marketing campaigns were based on phone calls. Often, more than one contact to the same client was required, in order to access if the product (bank term deposit) would be ('yes') or not ('no') subscribed. 

There are four datasets: 
1) bank-additional-full.csv with all examples (41188) and 20 inputs, ordered by date (from May 2008 to November 2010), very close to the data analyzed in [Moro et al., 2014]
2) bank-additional.csv with 10% of the examples (4119), randomly selected from 1), and 20 inputs.
3) bank-full.csv with all examples and 17 inputs, ordered by date (older version of this dataset with less inputs). 
4) bank.csv with 10% of the examples and 17 inputs, randomly selected from 3 (older version of this dataset with less inputs). 
The smallest datasets are provided to test more computationally demanding machine learning algorithms (e.g., SVM). 

3) What is the target?
- The  goal is to predict if the client will subscribe (yes/no) a term deposit (variable y).

4) Is this a classification or regression problem?
- This is a classification Problem
5) How many features?
- 17 Features

6) How many rows of data.
- 4521

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
- Cleaning
  - None
- Exploring
 - Correlation
 - Visualization 
- Preprocessing
  - Scaling
  - OneHotEncoding of Nominal Features

In [14]:
#loading dataset 2 Fertility Diagnosis
fertility_path = 'https://drive.google.com/uc?export=download&id=1LakqaJ2pBFM-K_7YoGlb4x3y7dCJZbZt'
fert_df = pd.read_csv(fertility_path, header=None)
fert_df.head()

0     1  2  3  4  5    6  7     8  9
0 -0.33  0.69  0  1  1  0  0.8  0  0.88  N
1 -0.33  0.94  1  0  1  0  0.8  1  0.31  O
2 -0.33  0.50  1  0  0  0  1.0 -1  0.50  N
3 -0.33  0.75  0  1  1  0  1.0 -1  0.38  N
4 -0.33  0.67  1  1  0  0  0.8 -1  0.50  O

In [15]:
fert_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 10 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   0       100 non-null    float64
 1   1       100 non-null    float64
 2   2       100 non-null    int64  
 3   3       100 non-null    int64  
 4   4       100 non-null    int64  
 5   5       100 non-null    int64  
 6   6       100 non-null    float64
 7   7       100 non-null    int64  
 8   8       100 non-null    float64
 9   9       100 non-null    object 
dtypes: float64(4), int64(5), object(1)
memory usage: 7.9+ KB


# Fertility Data Set

1) Source of data
- David Gil, 
dgil '@' dtic.ua.es, 
Lucentia Research Group, Department of Computer Technology, University of Alicante 

2) Brief description of data
-  100 volunteers provide a semen sample analyzed according to the WHO 2010 criteria. Sperm concentration are related to socio-demographic data, environmental factors, health status, and life habits

3) What is the target?
- Diagnosis normal (N), altered (O) 

4) Is this a classification or regression problem?
- This is Classificatio Problem

5) How many features?
- 10

6) How many rows of data.
- 100

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
- Cleaning
  - Renaming Columns
- Exploring
 - Correlation
 - Visualization 
- Preprocessing
  - Scaling


Attribute Information:
- Season in which the analysis was performed. 1) winter, 2) spring, 3) Summer, 4) fall. (-1, -0.33, 0.33, 1) 
- Age at the time of analysis. 18-36 (0, 1)
- Childish diseases (ie , chicken pox, measles, mumps, polio) 1) yes, 2) no. (0, 1)
- Accident or serious trauma 1) yes, 2) no. (0, 1)
- Surgical intervention 1) yes, 2) no. (0, 1)
- High fevers in the last year 1) less than three months ago, 2) more than three months ago, 3) no. (-1, 0, 1)
- Frequency of alcohol consumption 1) several times a day, 2) every day, 3) several times a week, 4) once a week, 5) hardly ever or never (0, 1) 
- Smoking habit 1) never, 2) occasional 3) daily. (-1, 0, 1)
- Number of hours spent sitting per day ene-16 (0, 1)
- Output: Diagnosis normal (N), altered (O)